tests/thandle_tuple_array_ut/thandle_tuple_array_ut.c

```diff
@@ -0,0 +1,479 @@
+
+    real_gballoc_hl_deinit();
+}
+
+TEST_FUNCTION_INITIALIZE(method_init)
+{
+    if (TEST_MUTEX_ACQUIRE(test_serialize_mutex))
+    {
+        ASSERT_FAIL("Could not acquire test serialization mutex.");
+    }
+
+    umock_c_reset_all_calls();
+    umock_c_negative_tests_init();
+}
+
+TEST_FUNCTION_CLEANUP(method_cleanup)
+{
+    umock_c_negative_tests_deinit();
+    TEST_MUTEX_RELEASE(test_serialize_mutex);
+}
+
+//
+// THANDLE_TUPLE_ARRAY_CREATE(name)
+//
+
+/*Tests_SRS_THANDLE_TUPLE_ARRAY_42_001: [ THANDLE_TUPLE_ARRAY_CREATE(name) shall allocate memory for the array. ]*/
+/*Tests_SRS_THANDLE_TUPLE_ARRAY_42_005: [ THANDLE_TUPLE_ARRAY_CREATE(name) shall succeed and return the allocated array. ]*/
+TEST_FUNCTION(THANDLE_TUPLE_ARRAY_CREATE_with_1_field_0_elements_succeeds)
+{
+    // arrange
+    STRICT_EXPECTED_CALL(malloc(IGNORED_ARG));
+
+    // act
+    THANDLE_TUPLE_ARRAY_TYPE(TUPLE_ONE)* result = THANDLE_TUPLE_ARRAY_CREATE(TUPLE_ONE)(0);
+
+    // assert
+    ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls());
+    ASSERT_IS_NOT_NULL(result);
+
+    // cleanup
+    THANDLE_TUPLE_ARRAY_DESTROY(TUPLE_ONE)(result);
+}
+
+/*Tests_SRS_THANDLE_TUPLE_ARRAY_42_001: [ THANDLE_TUPLE_ARRAY_CREATE(name) shall allocate memory for the array. ]*/
+/*Tests_SRS_THANDLE_TUPLE_ARRAY_42_005: [ THANDLE_TUPLE_ARRAY_CREATE(name) shall succeed and return the allocated array. ]*/
+TEST_FUNCTION(THANDLE_TUPLE_ARRAY_CREATE_with_3_fields_0_elements_succeeds)
+{
+    // arrange
+    STRICT_EXPECTED_CALL(malloc(IGNORED_ARG));
+
+    // act
+    THANDLE_TUPLE_ARRAY_TYPE(TUPLE_THREE)* result = THANDLE_TUPLE_ARRAY_CREATE(TUPLE_THREE)(0);
+
+    // assert
+    ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls());
+    ASSERT_IS_NOT_NULL(result);
+
+    // cleanup
+    THANDLE_TUPLE_ARRAY_DESTROY(TUPLE_THREE)(result);
+}
+
+/*Tests_SRS_THANDLE_TUPLE_ARRAY_42_001: [ THANDLE_TUPLE_ARRAY_CREATE(name) shall allocate memory for the array. ]*/
+/*Tests_SRS_THANDLE_TUPLE_ARRAY_42_002: [ THANDLE_TUPLE_ARRAY_CREATE(name) shall allocate memory for count structs which hold the tuple. ]*/
+/*Tests_SRS_THANDLE_TUPLE_ARRAY_42_003: [ THANDLE_TUPLE_ARRAY_CREATE(name) shall initialize the members of the tuples in the array to NULL. ]*/
+/*Tests_SRS_THANDLE_TUPLE_ARRAY_42_005: [ THANDLE_TUPLE_ARRAY_CREATE(name) shall succeed and return the allocated array. ]*/
+TEST_FUNCTION(THANDLE_TUPLE_ARRAY_CREATE_with_1_field_1_element_succeeds)
+{
+    // arrange
+    STRICT_EXPECTED_CALL(malloc(IGNORED_ARG));
+    STRICT_EXPECTED_CALL(malloc(sizeof(TUPLE_ONE)));
+    STRICT_EXPECTED_CALL(THANDLE_INITIALIZE(RC_STRING)(IGNORED_ARG, NULL));
+
+    // act
+    THANDLE_TUPLE_ARRAY_TYPE(TUPLE_ONE)* result = THANDLE_TUPLE_ARRAY_CREATE(TUPLE_ONE)(1);
+
+    // assert
+    ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls());
+    ASSERT_IS_NOT_NULL(result);
+    ASSERT_ARE_EQUAL(uint32_t, 1, result->count);

```

Comment:
>ASSERT_ARE_EQUAL [](http://example.com/codeflow?start=4&length=16)

missing assert that the element of the array is NULL (per SRS_THANDLE_TUPLE_ARRAY_42_003)